server-side/controllers/user.controller.js

```diff
@@ -44,7 +79,65 @@ export const addUser = async (req, res) => {
     res.status(500).send('Error saving user');
   }
 };
+export const signIn = async (req, res, next) => {
+  try {
+    console.log(req.body,"klkl");
+    const { email, password } = req.body;
+    const user = await User.findOne({ email });
+    if (user) {
+      console.log('Password from request:', password);
+      console.log('Password from database:', user.password);
+      
+      bcrypt.compare( password, user.password, (err, same) => {
+        if (err) {
+          console.log("Error in bcrypt compare:", err);
+          return next(new Error(err.message));
+        }
+
+        if (same) {
+          user.password = "****";
+          const token=generateToken(user);
+          console.log("Password match successful");
+          return res.send({ user,token });
+        } else {
+          console.log("Password does not match");
+          return res.status(401).send({ message: 'Auth Failed' });
```

Comment:
it's not secured to return password don't match, it should be something like: user or password doesn't exists or not match